### PR TITLE
[BB-1713] Fix actions bug

### PR DESCRIPTION
**Pull request by mateovespConductor**
#### Description

- [x] Bug fix
- [ ] New feature

#### Useful links:

- [Baton SDK coding guidelines](https://github.com/ConductorOne/baton-sdk/wiki/Coding-Guidelines)
- [New contributor guide](https://github.com/ConductorOne/baton/blob/main/CONTRIBUTING.md)

## Summary by CodeRabbit

* **New Features**
  * Added a new endpoint to fetch individual users by ID.

* **Bug Fixes**
  * Improved URL encoding of user identifiers across endpoints.
  * Enhanced user activation logic to skip unnecessary updates when users are already active.

* **Tests**
  * Added comprehensive test coverage for the new user fetch functionality and related edge cases.

### CODE REVIEW

File: pkg/connector/actions.go [hunk 1]
```diff
@@ -108,27 +110,45 @@ func (c *Connector) EnableUser(ctx context.Context, args *structpb.Struct) (*str
...
+		if st, ok := status.FromError(err); ok && st.Code() == codes.NotFound {
+			l.Info("GetUserByID returned 404: endpoint does not return inactive/disabled users; this is expected behavior when the user is disabled in Metabase",
+				zap.String("userId", userIdStr))
+			success = false
+		} else {
+			return nil, ann, fmt.Errorf("failed to fetch user %s: %w", userIdStr, err)
+		}
+	} else {
+		success = user.IsActive
 	}
```
[the remark this answers is not in the document]
Comment (mateovespConductor) [reply pkg/connector/actions.go:129]: GetUserByID returns a 404 HTTP error wrapped by the Baton HTTP client, which is properly handled through the gRPC status translation.
The status.FromError(err) check actually works correctly — you can see it logs the expected warning when the user is disabled (Metabase's /user/{id} endpoint does not return inactive users).
I verified this manually, and the enable flow completes successfully.

**Review by luisina-santos — APPROVED** [review]

File: pkg/connector/actions.go [hunk 2]
```diff
@@ -108,27 +110,45 @@ func (c *Connector) EnableUser(ctx context.Context, args *structpb.Struct) (*str
...
+		user, rateLimitDesc, err = c.client.UpdateUserActiveStatus(ctx, userIdStr, true)
+		if rateLimitDesc != nil {
+			ann.WithRateLimiting(rateLimitDesc)
+		}
+		if err != nil {
+			return nil, ann, fmt.Errorf("failed to enable user %s: %w", userIdStr, err)
+		}
+		success = user.IsActive
+		if !success {
+			l.Warn("user enable operation completed but user is still inactive", zap.String("userId", userIdStr), zap.Bool("active", user.IsActive))
```
Comment (agustin-conductor) [inline pkg/connector/actions.go:142]: do we want to return an error here? I'm not entirely sure c1 checks the success value to evaluate  (it wans't before, don't know if something change) whether the action failed or succeeded.  I think it only errors if we return an error?

**Review by agustin-conductor — APPROVED** [review]
Left a comment about the edge case where the user remains disabled even when the api call succeeded.

File: pkg/connector/actions.go (hunk 2 above)
Comment (luisina-santos) [reply pkg/connector/actions.go:142]: this is consistent with existent lifecycle actions implementations, so we should work on confirm what's the best pattern
